main/src/library/StringBuilder.flix

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2019 Stephen Tetley
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+///
+/// Represents a StringBuilder.
+///
+opaque type StringBuilder = ##flix.runtime.library.MonoStringBuilder
+
+namespace StringBuilder {
+
+
+    ///
+    /// Returns a new mutable StringBuilder.
+    ///
+    pub def new(): StringBuilder =
+        let msb = native new flix.runtime.library.MonoStringBuilder();
+        StringBuilder(msb)
+
+    ///
+    /// Convert the StringBuilder `sb` to a string.
+    ///
+    pub def toString(sb: StringBuilder): Str =
+        let StringBuilder(msb) = sb;
+        native method flix.runtime.library.MonoStringBuilder.toString(msb) as Str
+
+    ///
+    /// Append the StringBuilder `sb1` to the end of StringBuilder `sb`.
+    ///
+    pub def append(sb: StringBuilder, sb1: StringBuilder): Unit =
+        let StringBuilder(msb) = sb;
+        let s1: Str = toString(sb1);
+        native method flix.runtime.library.MonoStringBuilder.appendString(msb, s1) as Unit
+
+    ///
+    /// Alias for `appendString`
+    ///
+    pub def appendStr(sb: StringBuilder, s: Str): Unit = appendString(sb, s)
+
+    ///
+    /// Append the Str `s` to the StringBuffer `sb`.
+    ///
+    pub def appendString(sb: StringBuilder, s: Str): Unit =

```

Comment:
In the future, I am planning to rename `Str` to `String` so having two methods `appendStr` and `appendString` might be confusing. Maybe we could call one of  them `appendToBuffer` or the like? I am open to suggestions.